mmv1/third_party/terraform/tests/resource_iam_deny_policy_test.go.erb

```diff
@@ -0,0 +1,197 @@
+		Providers:    testAccProvidersOiCS,
+		CheckDestroy: testAccCheckIAM2DenyPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIAM2DenyPolicy_iamDenyPolicyFolder(context),
+			},
+			{
+				ResourceName:            "google_iam_deny_policy.example",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"name", "parent"},
+			},
+		},
+	})
+}
+
+func testAccIAM2DenyPolicy_iamDenyPolicyUpdate(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_project" "project" {
+  provider        = google-beta
+  project_id      = "tf-test%{random_suffix}"
+  name            = "tf-test%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+resource "google_iam_deny_policy" "example" {
+  provider = google-beta
+  parent   = urlencode("cloudresourcemanager.googleapis.com/projects/${google_project.project.project_id}")
+  name     = "tf-test-my-deny-policy%{random_suffix}"
+  display_name = "A deny rule"
+  rules {
+    description = "First rule"
+    deny_rule {
+      denied_principals = ["principal://iam.googleapis.com/projects/-/serviceAccounts/${google_service_account.test-account.email}"]
+      denial_condition {
+        title = "Some expr"
+        expression = "!resource.matchTag('12345678/env', 'test')"
+      }
+      denied_permissions = ["cloudresourcemanager.googleapis.com/projects.delete"]
+    }
+  }
+  rules {
+    description = "Second rule"
+    deny_rule {
+      denied_principals = ["principalSet://goog/public:all"]
+      denial_condition {
+        title = "Some expr"
+        expression = "!resource.matchTag('12345678/env', 'test')"
+      }
+      denied_permissions = ["cloudresourcemanager.googleapis.com/projects.delete"]
+      exception_principals = ["principal://iam.googleapis.com/projects/-/serviceAccounts/${google_service_account.test-account.email}"]
+    }
+  }
+}
+
+resource "google_service_account" "test-account" {
+  provider = google-beta
+  account_id   = "tf-test-deny-account%{random_suffix}"
+  display_name = "Test Service Account"
+  project      = google_project.project.project_id
+}
+`, context)
+}
+
+func testAccIAM2DenyPolicy_iamDenyPolicyUpdate2(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_project" "project" {
+  provider        = google-beta
+  project_id      = "tf-test%{random_suffix}"
+  name            = "tf-test%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+resource "google_iam_deny_policy" "example" {
+  provider = google-beta
+  parent   = urlencode("cloudresourcemanager.googleapis.com/projects/${google_project.project.project_id}")
+  name     = "tf-test-my-deny-policy%{random_suffix}"
+  display_name = "A deny rule"
+  rules {
+    description = "Second rule"
+    deny_rule {
+      denied_principals = ["principalSet://goog/public:all"]
+      denial_condition {
```

Comment:
Can we update some subvalues in here? I don't think location / description are used, and title could be updated.